src/primitive/time_evolution.jl

```diff
@@ -33,20 +32,14 @@ Optional keywords are tolerance `tol` (default is `1e-7`)
 `TimeEvolution` block can also be used for
 [imaginary time evolution](http://large.stanford.edu/courses/2008/ph372/behroozi2/) if dt is complex.
 """
-TimeEvolution(H::AbstractBlock, dt; tol::Real=1e-7, check_hermicity=true) =
-    TimeEvolution(BlockMap(H), dt, tol, check_hermicity=check_hermicity)
+TimeEvolution(H::AbstractBlock, dt; tol::Real=1e-7) = TimeEvolution(mat(H), dt, tol)
+TimeEvolution(M::AbstractMatrix, dt; tol::Real) = TimeEvolution(M, dt, tol)
 
-TimeEvolution(M::BlockMap, dt; tol::Real, check_hermicity=true) =
-    TimeEvolution(M, dt, tol, check_hermicity=check_hermicity)
-
-time_evolve(M::BlockMap, dt; kwargs...) = TimeEvolution(M, dt; kwargs...)
+time_evolve(M::AbstractMatrix, dt; kwargs...) = TimeEvolution(M, dt; kwargs...)
```

Comment:
Please don't use matrix in quantum computing, always use operators. Otherwise a disaster for device control, compiling et. al.